specs/labels-and-localization/README.md

```diff
@@ -0,0 +1,188 @@
+- Expand/Collapse Group button, Collapse All buttons: These are already `nimble-button` with `content-hidden`, but currently contain no text. We need to add text content for them (which will need localization).
+- Action menu button in table cells ([#859](https://github.com/ni/nimble/issues/859)): Base table column already has `action-menu-label` for the accessible label (for which clients would handle localizing themselves). Currently providing `action-menu-label` is not required, so we may want to define a table fallback string to use if it's not provided (e.g. "Configure" or "Options") which would need localization.
+- Column menu (button in column header): Will need a Nimble-provided label (like "Column Options") which will need localization. Each menu item will be both an icon and a visible label/ menu text, so no `title`/label specifically on the icons in the menu items is needed. Each menu item's text will be Nimble-provided (and need localization).
+- Sort indicators (ascending/descending icon) in table headers: No accessible label specifically for these icons is currently planned. The primary sorted column is already indicated with `aria-sort="ascending"` or `aria-sort="descending"`. The [ARIA APG's sortable table example](https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/) follows similar logic, and says sort labels are not added to each column header button "to prevent repetitious verbosity that could interfere with understanding of the column titles".
+- Client-provided icon elements in the table (e.g. icons as the primary content of table headers): Follows the same guidance in [accessibility.md](../../packages/nimble-components/docs/accessibility.md), i.e. the icons should provide accessible text via the `title` attribute (and clients will handle localizing it themselves).
+- (Mapping/ Icon Columns: As currently specced, `nimble-table-column-icon`'s `nimble-mapping-icon` has a `label` attribute which will become the icon `title`)
+
+## Localization
+
+Our general approach to localization is that clients will handle localizing strings. That has led us to add attributes on some Nimble components for clients to provide localized labels, but that doesn't scale well to components with many labels like the table.
+
+### Plan for Nimble-Provided Labels
+
+Conceptually the localized strings are very similar to the sharing pattern of Design Tokens. The vast majority of the time, you want to use the same value (i.e. numeric increment/decrement button text or collapse button text is not control instance specific) but want to adjust to a global config, i.e. theme or language. But also want to be able to override for specific controls as needed.
+
+We will create non-CSS-property design tokens, similar to the `theme` and `direction` design tokens. The values can be set via the `nimble-theme-provider`:
+
+```html
+<nimble-theme-provider
+    label-number-field-decrement="Increment"
```

Comment:
My take on @jattasNI questions:
1. Yes, we should still call it `label` (or some other facsimile, but "label" seems fairly ubiquitous). Otherwise the attribute semantic meaning would be unclear (i.e. what would a user assume `number-field-decrement` was meant to represent).
2. I agree with the alternate name.

Reply:
Yes the `label-` prefix was mainly to distinguish it from other theme provider attributes.

I guess I don't have strong justification for having it at the beginning - it made sense to me in terms of having the names get more specific as you go (`label-*` is all labels, `label-number-field-*` is all number field labels, etc). But I'm fine with having it at the end if that's the general preference.

I'll add a section describing the name format once a few more people comment with their preference.

Reply:
Added a section on the naming scheme, moved `-label` to the end.